src/Selenium2Driver.php

```diff
@@ -278,6 +278,24 @@ private function executeJsOnElement(Element $element, $script, $sync = true)
         return $this->wdSession->execute_async($options);
     }
 
+    /**
+     * Execute Javascript code on element to dispatch event.
+     *
+     * @param string $event HTMLEvents
+     * @param string $xpath Xpath of the element
+     */
+    public function dispatchJSEventOnElement($event, $xpath)
+    {
+        $script = <<<JS
+(function (element) {
+var event = document.createEvent("HTMLEvents");
+event.initEvent("{$event}", true, true);
+element.dispatchEvent(event);
+}({{ELEMENT}}));
+JS;
+        $this->withSyn()->executeJsOnXpath($xpath, $script);
+    }
```

Comment:
Please replace the similar code in `dragTo()` with calls to this new method:

## Before

```php
    public function dragTo($sourceXpath, $destinationXpath)
    {
        $source      = $this->findElement($sourceXpath);
        $destination = $this->findElement($destinationXpath);

        $this->wdSession->moveto(array(
            'element' => $source->getID()
        ));

        $script = <<<JS
(function (element) {
    var event = document.createEvent("HTMLEvents");
    event.initEvent("dragstart", true, true);
    event.dataTransfer = {};
    element.dispatchEvent(event);
}({{ELEMENT}}));
JS;
        $this->withSyn()->executeJsOnElement($source, $script);

        $this->wdSession->buttondown();
        $this->wdSession->moveto(array(
            'element' => $destination->getID()
        ));
        $this->wdSession->buttonup();

        $script = <<<JS
(function (element) {
    var event = document.createEvent("HTMLEvents");
    event.initEvent("drop", true, true);
    event.dataTransfer = {};
    element.dispatchEvent(event);
}({{ELEMENT}}));
JS;
        $this->withSyn()->executeJsOnElement($destination, $script);
    }
```

## After

```php
    public function dragTo($sourceXpath, $destinationXpath)
    {
        $destination = $this->findElement($destinationXpath);

        $this->wdSession->moveto(array(
            'element' => $source->getID()
        ));

        $this->dispatchJSEventOnElement('dragstart', $sourceXpath);
        
        $this->wdSession->buttondown();
        $this->wdSession->moveto(array(
            'element' => $destination->getID()
        ));
        $this->wdSession->buttonup();

        $this->dispatchJSEventOnElement('drop', $destinationXpath);
    }
```

Reply:
I knew, i forget something..

Reply:
Good idea 👍